### ISSUE TITLE
Is the model equal to a (carefully special) pre-trained embedding + ConvKB?

### ISSUE THREAD

**Issue by iamlockelightning** [deepakn97/relationPrediction#14]
Thanks for sharing your code. I am really impressed by this paper. I find that this encoder-decoder model works like:
```python3
train_gat(args)     # train the (multi-hop neighbor...) GAT with a TransE loss, and save the embedding

train_conv(args)    # use the saved embedding as input to train ConvKB for the link prediction task
```
So there is no other interaction between the GAT model and the ConvKB model except for the trained embeddings,  am I right?

If so, is the trained embeddings fixed in ConvKB or trainable?

Thank you and hope for your reply.

**Comment by chauhanjatin10** [comment]
Hi @iamlockelightning . Thanks for showing interest in our work. You are exactly correct. This model introduces an inductive bias by pre-training the entity and relation embeddings for capturing graph-structure and rich semantic information. While training the ConvKB model, we tried both the ways - (a) by fixing the trained embeddings in the ConvKB model as well as (b)  keeping them trainable parameters. We observed slightly better results by doing the latter. I hope this clarifies. We observed that training the whole model end to end was quite time consuming due to higher variance in training as well as vanishing gradients in the GAT. Hence, we trained these models separately.
Thanks once again.
